documentation/CallGraph.md

```diff
@@ -0,0 +1,287 @@
+that path is preserved through the creation of a new edge connecting the included
+nodes. Expected configuration format is `"Fold"`.
+3. `Deduplicate`: This reduction reduces the set of edges in the graph such that
+each there exists at most one edge connecting any two nodes. This has the effect of
+removing type information from the graph, and makes it more suitable for
+visualization. Expected configuration format is `"Deduplicate"`.
+4. `Clean`: This reduction removes *orphan nodes* from the graph. These are nodes
+that have no incoming or outgoing edges. Expected configuration format is `"Clean"`.
+
+Reductions may be specified in any order (and even multiple times), and they are
+applied sequentially in the order specified.
+
+## Dot output
+
+The call graph generator supports dot output of the call graph. This output format
+is intended for use with [Graphviz](https://graphviz.org/), which ingests the dot
+file format to produce a visualization of the graph in multiple formats.
+
+In this form nodes are labeled with their corresponding function name while edges
+are not labeled. It is recommended to use the `Deduplicate` reduction to remove
+extra edges between nodes.
+
+Here is an example of the dot output:
+```
+digraph {
+    0 [ label = "\"static[8787]::main\"" ]
+    1 [ label = "\"static[8787]::fn1\"" ]
+    2 [ label = "\"static[8787]::fn2\"" ]
+    3 [ label = "\"static[8787]::fn3\"" ]
+    0 -> 1 [ ]
+    1 -> 2 [ ]
+    2 -> 3 [ ]
+}
+```
+
+Supposing this is stored in a file called `graph.dot`, this can be
+used to produce a PDF of the graph using the graphviz `dot` tool:
+```
+$ dot -Tpdf graph.dot -o graph.pdf
+```
+
+## Datalog output
+
+The call graph generator also supports 
+[Datalog](https://en.wikipedia.org/wiki/Datalog) output of the call graph.
+Datalog is a logic programming language that can be used, in conjunction with the
+generated call graph, to write an analysis that operates on the call graph.
+
+Here is an example of the Datalog output for the prior graph:
+```
+start;
+insert Edge(0,0,1);
+insert Edge(1,1,2);
+insert Edge(2,2,3);
+insert EdgeType(0,0);
+insert EdgeType(1,0);
+insert EdgeType(2,0);
+commit;
+```
+
+The call graph generator generates two types of relations:
+1. Graph relations: Specify properties about the graph itself.
+2. Type relations: Specify properties about the types of the edges of the graph.
+
+These relations are explained in detail below.
+
+Note: To preserve type information it is recommended to **not** use the
+`Deduplicate` reduction.
+
+### Graph relations
+
+There are three graph relations output by MIRAI:
+1. `Edge(id,n1,n2)`: This specifies that there is a directed (call) edge from
+`n1` to `n2`. The edge has the identifier `id`.
+2. `EdgeType(id,t)`: This specifies that the edge with the identifier `id` has
+type `t`.
+3. `Dom(n1,n2)`: This specifies that call to node `n1` dominates the call to node
```

Comment:
You may want to explain exactly what "dominates" means in this context.

Reply:
Got it, I will clear that up.